crates/context_aware_config/src/api/context/types.rs

```diff
@@ -5,11 +5,18 @@ use serde_json::{Map, Value};
 pub struct PutReq {
     pub context: Map<String, Value>,
     pub r#override: Map<String, Value>,
+    pub tags: Option<Vec<String>>,
```

Comment:
Can we accept tags as headers? These don't really belong here, and the name is also misleading should be `snapshot_tags`

Reply:
headers can be used when its value is consistent across similar requests or let say it is a global level identifier for requests . 
although tags is consistent across all write requests , I think we should not add this in header because probably tags can be changed based on the different operations (c, u, d) on different entities (default_configs, contexts)

also we have other entities and request where this header is no need ( like functions, types, get apis ) so for me it probably seems more close to a create request than a global request identifier 
will change tags -> version_tags 

@knutties @Datron your thoughts ?

Reply:
`tags` is a actually a metadata here, and not exactly the part of the request, are we using this to tag the actual resource we are touching in these request, the answer is no, these tags in fact are used to tag the versions created after the actual defined job for the request is completed.

Reply:
We are repeating everywhere too, all request types are changed, can't we do something to abstract this out.

Reply:
added tags in headers